// autumn-js/dist/libraries/backend/utils/secretKeyCheck.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/backend/utils/secretKeyCheck.ts
var secretKeyCheck_exports = {};
__export(secretKeyCheck_exports, {
  secretKeyCheck: () => secretKeyCheck
});
module.exports = __toCommonJS(secretKeyCheck_exports);
var secretKeyCheck = (secretKey) => {
  if (!secretKey && !process.env.AUTUMN_SECRET_KEY) {
    return {
      found: false,
      error: {
        statusCode: 500,
        message: `Autumn secret key not found in ENV variables or passed into autumnHandler`,
        code: "no_secret_key"
      }
    };
  }
  return { found: true, error: null };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  secretKeyCheck
});
