// autumn-js/dist/libraries/backend/schemas/zod.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/backend/schemas/zod.ts
var zod_exports = {};
__export(zod_exports, {
  attachFeatureOptionsSchema: () => attachFeatureOptionsSchema,
  attachParamsSchema: () => attachParamsSchema,
  cancelParamsSchema: () => cancelParamsSchema,
  checkParamsSchema: () => checkParamsSchema,
  createReferralCodeParamsSchema: () => createReferralCodeParamsSchema,
  entityDataParamsSchema: () => entityDataParamsSchema,
  openBillingPortalParamsSchema: () => openBillingPortalParamsSchema,
  redeemReferralCodeParamsSchema: () => redeemReferralCodeParamsSchema,
  trackParamsSchema: () => trackParamsSchema
});
module.exports = __toCommonJS(zod_exports);
var import_v4 = require("zod/v4");
var entityDataParamsSchema = import_v4.z.object({
  name: import_v4.z.string().optional(),
  featureId: import_v4.z.string()
});
var attachFeatureOptionsSchema = import_v4.z.object({
  featureId: import_v4.z.string(),
  quantity: import_v4.z.number()
});
var attachParamsSchema = import_v4.z.object({
  productId: import_v4.z.string().optional(),
  entityId: import_v4.z.string().optional(),
  options: import_v4.z.array(attachFeatureOptionsSchema).optional(),
  productIds: import_v4.z.array(import_v4.z.string()).optional(),
  freeTrial: import_v4.z.boolean().optional(),
  successUrl: import_v4.z.string().optional(),
  metadata: import_v4.z.record(import_v4.z.string(), import_v4.z.string()).optional(),
  forceCheckout: import_v4.z.boolean().optional(),
  entityData: entityDataParamsSchema.optional(),
  openInNewTab: import_v4.z.boolean().optional(),
  reward: import_v4.z.string().optional(),
  checkoutSessionParams: import_v4.z.record(import_v4.z.string(), import_v4.z.any()).optional(),
  newBillingSubscription: import_v4.z.boolean().optional()
});
var cancelParamsSchema = import_v4.z.object({
  productId: import_v4.z.string(),
  entityId: import_v4.z.string().optional(),
  cancelImmediately: import_v4.z.boolean().optional()
});
var checkParamsSchema = import_v4.z.object({
  featureId: import_v4.z.string().optional(),
  productId: import_v4.z.string().optional(),
  entityId: import_v4.z.string().optional(),
  requiredBalance: import_v4.z.number().optional(),
  sendEvent: import_v4.z.boolean().optional(),
  withPreview: import_v4.z.boolean().optional(),
  entityData: entityDataParamsSchema.optional()
});
var trackParamsSchema = import_v4.z.object({
  featureId: import_v4.z.string().optional(),
  eventName: import_v4.z.string().optional(),
  entityId: import_v4.z.string().optional(),
  value: import_v4.z.number().optional(),
  idempotencyKey: import_v4.z.string().optional(),
  entityData: entityDataParamsSchema.optional()
});
var openBillingPortalParamsSchema = import_v4.z.object({
  returnUrl: import_v4.z.string().optional(),
  openInNewTab: import_v4.z.boolean().optional()
});
var createReferralCodeParamsSchema = import_v4.z.object({
  programId: import_v4.z.string()
});
var redeemReferralCodeParamsSchema = import_v4.z.object({
  code: import_v4.z.string()
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  attachFeatureOptionsSchema,
  attachParamsSchema,
  cancelParamsSchema,
  checkParamsSchema,
  createReferralCodeParamsSchema,
  entityDataParamsSchema,
  openBillingPortalParamsSchema,
  redeemReferralCodeParamsSchema,
  trackParamsSchema
});
