// autumn-js/dist/libraries/react/utils/errorUtils.js
"use client";

"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/react/utils/errorUtils.ts
var errorUtils_exports = {};
__export(errorUtils_exports, {
  AutumnClientError: () => AutumnClientError,
  toClientError: () => toClientError
});
module.exports = __toCommonJS(errorUtils_exports);
var AutumnClientError = class extends Error {
  constructor(error) {
    super(error.message);
    this.error = error;
    this.code = error.code;
  }
  code;
  toString() {
    return `${this.message} (${this.code})`;
  }
  toJSON() {
    return { message: this.message, code: this.code };
  }
};
var toClientError = (error) => {
  let msg = "Unknown error";
  let code = "unknown";
  if (error?.message) {
    msg = error.message;
  }
  if (error?.code) {
    code = error.code;
  }
  return {
    data: null,
    error: new AutumnClientError({ message: msg, code })
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AutumnClientError,
  toClientError
});
