// autumn-js/dist/libraries/react/utils/inject-styles.js
"use client";

"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/react/utils/inject-styles.ts
var inject_styles_exports = {};
__export(inject_styles_exports, {
  loadingStyles: () => loadingStyles,
  spinnerStyles: () => spinnerStyles
});
module.exports = __toCommonJS(inject_styles_exports);
var loadingStyles = {
  width: "100%",
  height: "100%",
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  minHeight: "300px"
};
var spinnerStyles = {
  width: "1.5rem",
  height: "1.5rem",
  color: "rgb(161 161 170)",
  animation: "spin 1s linear infinite"
};
if (typeof document !== "undefined") {
  const styleId = "au-spinner-keyframes";
  if (!document.getElementById(styleId)) {
    const style = document.createElement("style");
    style.id = styleId;
    style.textContent = `
      @keyframes spin {
        from { transform: rotate(0deg); }
        to { transform: rotate(360deg); }
      }
    `;
    document.head.appendChild(style);
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  loadingStyles,
  spinnerStyles
});
