// autumn-js/dist/libraries/backend/utils/betterAuth/middlewares.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/backend/utils/betterAuth/middlewares.ts
var middlewares_exports = {};
__export(middlewares_exports, {
  getIdentityContext: () => getIdentityContext,
  getOrganizationContext: () => getOrganizationContext,
  scopeContainsOrg: () => scopeContainsOrg
});
module.exports = __toCommonJS(middlewares_exports);
var import_api = require("better-auth/api");
var scopeContainsOrg = ({ options }) => {
  return options?.customerScope === "organization" || options?.customerScope === "user_and_organization";
};
async function getOrganizationContext(ctx, options) {
  if (!scopeContainsOrg({ options }) && !options?.identify) {
    return {
      activeOrganizationId: null,
      activeOrganization: null,
      activeOrganizationEmail: null
    };
  }
  try {
    const session = await (0, import_api.getSessionFromCtx)(ctx);
    const orgId = session?.session.activeOrganizationId;
    if (orgId && session) {
      const [member, organization] = await Promise.all([
        ctx.context.adapter.findOne({
          model: "member",
          where: [
            { field: "userId", value: session.user.id },
            { field: "organizationId", value: orgId }
          ]
        }),
        ctx.context.adapter.findOne({
          model: "organization",
          where: [{ field: "id", value: orgId }]
        })
      ]);
      if (!member) {
        return {
          activeOrganizationId: null,
          activeOrganization: null,
          activeOrganizationEmail: null
        };
      }
      const creatorRole = ctx.context.orgOptions?.creatorRole || "owner";
      let ownerUserId = member.role === creatorRole ? member.userId : null;
      let owner = null;
      if (member.role !== creatorRole) {
        const ownerMembers = await ctx.context.adapter.findMany({
          model: "member",
          where: [
            { field: "organizationId", value: orgId },
            { field: "role", value: creatorRole }
          ]
        });
        ownerUserId = ownerMembers.length > 0 ? ownerMembers[0].userId : null;
      }
      if (ownerUserId) {
        const ownerUser = await ctx.context.adapter.findOne({
          model: "user",
          where: [{ field: "id", value: ownerUserId }]
        });
        owner = ownerUser;
      }
      return {
        activeOrganizationId: orgId,
        activeOrganization: organization,
        activeOrganizationEmail: owner?.email
      };
    }
    return {
      activeOrganizationId: null,
      activeOrganization: null,
      activeOrganizationEmail: null
    };
  } catch (error) {
    console.log("[org context error]", error);
    return {
      activeOrganizationId: null,
      activeOrganization: null,
      activeOrganizationEmail: null
    };
  }
}
async function getIdentityContext({
  orgContext,
  options,
  session
}) {
  if (!options?.identify) return null;
  const orgData = orgContext.activeOrganization?.id ? {
    ...orgContext.activeOrganization,
    ownerEmail: orgContext.activeOrganizationEmail
  } : null;
  try {
    const identity = await options.identify({
      session,
      organization: orgData
    });
    return identity;
  } catch (error) {
    console.log("[identity context error]", error);
    return null;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getIdentityContext,
  getOrganizationContext,
  scopeContainsOrg
});
