// autumn-js/dist/libraries/react/chunk-6EXNYWJX.mjs
"use client";


// src/libraries/react/client/clientGenMethods.ts
async function checkoutMethod(params) {
  const finalParams = {
    ...params,
    successUrl: params.successUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/checkout`, finalParams);
  return res;
}
async function attachMethod(params) {
  const finalParams = {
    ...params,
    successUrl: params.successUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/attach`, finalParams);
  return res;
}
async function setupPaymentMethod(params = {}) {
  const finalParams = {
    ...params,
    successUrl: params.successUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/setup_payment`, finalParams);
  return res;
}
async function cancelMethod(params) {
  const res = await this.post(`${this.prefix}/cancel`, params);
  return res;
}
async function checkMethod(params) {
  const noDialogParams = {
    ...params,
    dialog: void 0
  };
  const res = await this.post(`${this.prefix}/check`, noDialogParams);
  return res;
}
async function trackMethod(params) {
  const res = await this.post(`${this.prefix}/track`, params);
  return res;
}
async function openBillingPortalMethod(params) {
  const finalParams = {
    ...params || {},
    returnUrl: params?.returnUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/billing_portal`, finalParams);
  return res;
}
async function queryMethod(params) {
  const res = await this.post(`${this.prefix}/query`, params);
  return res;
}

export {
  checkoutMethod,
  attachMethod,
  setupPaymentMethod,
  cancelMethod,
  checkMethod,
  trackMethod,
  openBillingPortalMethod,
  queryMethod
};
