// autumn-js/dist/libraries/react/client/types/clientAttachTypes.mjs
"use client";


// src/libraries/react/client/types/clientAttachTypes.ts
import { z } from "zod/v4";
var AttachFeatureOptionsSchema = z.object({
  featureId: z.string(),
  quantity: z.number()
});
var AttachParamsSchema = z.object({
  productId: z.string().optional(),
  entityId: z.string().optional(),
  options: z.array(AttachFeatureOptionsSchema).optional(),
  productIds: z.array(z.string()).optional(),
  freeTrial: z.boolean().optional(),
  successUrl: z.string().optional(),
  metadata: z.record(z.string(), z.string()).optional(),
  forceCheckout: z.boolean().optional(),
  /**
   * @deprecated This field is deprecated and will be removed in a future version.
   */
  dialog: z.any().optional().describe(
    "DEPRECATED: This field is deprecated and will be removed in a future version. Please use the checkout() method instead."
  ),
  entityData: z.any().optional(),
  openInNewTab: z.boolean().optional(),
  reward: z.string().optional(),
  checkoutSessionParams: z.record(z.string(), z.any()).optional(),
  newBillingSubscription: z.boolean().optional()
});
var CheckoutParamsSchema = z.object({
  productId: z.string().optional(),
  productIds: z.array(z.string()).optional(),
  entityId: z.string().optional(),
  entityData: z.any().optional(),
  options: z.array(AttachFeatureOptionsSchema).optional(),
  successUrl: z.string().optional(),
  openInNewTab: z.boolean().optional(),
  dialog: z.any().optional(),
  forceCheckout: z.boolean().optional(),
  freeTrial: z.boolean().optional(),
  checkoutSessionParams: z.record(z.string(), z.any()).optional(),
  reward: z.string().optional(),
  newBillingSubscription: z.boolean().optional()
});
export {
  AttachFeatureOptionsSchema,
  AttachParamsSchema,
  CheckoutParamsSchema
};
