// autumn-js/dist/libraries/react/client/types/clientEntTypes.js
"use client";

"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/react/client/types/clientEntTypes.ts
var clientEntTypes_exports = {};
__export(clientEntTypes_exports, {
  CreateEntityParamsSchema: () => CreateEntityParamsSchema,
  EntityDataParamsSchema: () => EntityDataParamsSchema,
  GetEntityParamsSchema: () => GetEntityParamsSchema
});
module.exports = __toCommonJS(clientEntTypes_exports);
var import_v4 = require("zod/v4");
var CreateEntityParamsSchema = import_v4.z.object({
  id: import_v4.z.string(),
  name: import_v4.z.string().optional(),
  featureId: import_v4.z.string()
});
var GetEntityParamsSchema = import_v4.z.object({
  expand: import_v4.z.array(import_v4.z.string()).optional()
});
var EntityDataParamsSchema = import_v4.z.object({
  name: import_v4.z.string().optional(),
  featureId: import_v4.z.string()
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CreateEntityParamsSchema,
  EntityDataParamsSchema,
  GetEntityParamsSchema
});
