// autumn-js/dist/libraries/react/client/types/clientGenTypes.mjs
"use client";

import {
  QueryRangeEnum
} from "../../chunk-QN2ALOVD.mjs";

// src/libraries/react/client/types/clientGenTypes.ts
import { z } from "zod/v4";
var CancelParamsSchema = z.object({
  productId: z.string(),
  entityId: z.string().optional(),
  cancelImmediately: z.boolean().optional()
});
var CheckParamsSchema = z.object({
  featureId: z.string().optional(),
  productId: z.string().optional(),
  entityId: z.string().optional(),
  requiredBalance: z.number().optional(),
  sendEvent: z.boolean().optional(),
  withPreview: z.boolean().optional(),
  dialog: z.any().optional(),
  entityData: z.any().optional(),
  properties: z.record(z.string(), z.any()).optional()
});
var TrackParamsSchema = z.object({
  featureId: z.string().optional(),
  eventName: z.string().optional(),
  entityId: z.string().optional(),
  value: z.number().optional(),
  idempotencyKey: z.string().optional(),
  entityData: z.any().optional()
});
var OpenBillingPortalParamsSchema = z.object({
  returnUrl: z.string().optional(),
  openInNewTab: z.boolean().optional()
});
var SetupPaymentParamsSchema = z.object({
  successUrl: z.string().optional(),
  checkoutSessionParams: z.record(z.string(), z.any()).optional(),
  openInNewTab: z.boolean().optional()
}).optional();
var QueryParamsSchema = z.object({
  featureId: z.string().or(z.array(z.string())),
  range: QueryRangeEnum.optional()
});
export {
  CancelParamsSchema,
  CheckParamsSchema,
  OpenBillingPortalParamsSchema,
  QueryParamsSchema,
  SetupPaymentParamsSchema,
  TrackParamsSchema
};
