// autumn-js/dist/libraries/react/client/clientGenMethods.js
"use client";

"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/libraries/react/client/clientGenMethods.ts
var clientGenMethods_exports = {};
__export(clientGenMethods_exports, {
  attachMethod: () => attachMethod,
  cancelMethod: () => cancelMethod,
  checkMethod: () => checkMethod,
  checkoutMethod: () => checkoutMethod,
  openBillingPortalMethod: () => openBillingPortalMethod,
  queryMethod: () => queryMethod,
  setupPaymentMethod: () => setupPaymentMethod,
  trackMethod: () => trackMethod
});
module.exports = __toCommonJS(clientGenMethods_exports);
async function checkoutMethod(params) {
  const finalParams = {
    ...params,
    successUrl: params.successUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/checkout`, finalParams);
  return res;
}
async function attachMethod(params) {
  const finalParams = {
    ...params,
    successUrl: params.successUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/attach`, finalParams);
  return res;
}
async function setupPaymentMethod(params = {}) {
  const finalParams = {
    ...params,
    successUrl: params.successUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/setup_payment`, finalParams);
  return res;
}
async function cancelMethod(params) {
  const res = await this.post(`${this.prefix}/cancel`, params);
  return res;
}
async function checkMethod(params) {
  const noDialogParams = {
    ...params,
    dialog: void 0
  };
  const res = await this.post(`${this.prefix}/check`, noDialogParams);
  return res;
}
async function trackMethod(params) {
  const res = await this.post(`${this.prefix}/track`, params);
  return res;
}
async function openBillingPortalMethod(params) {
  const finalParams = {
    ...params || {},
    returnUrl: params?.returnUrl ?? this.defaultReturnUrl
  };
  const res = await this.post(`${this.prefix}/billing_portal`, finalParams);
  return res;
}
async function queryMethod(params) {
  const res = await this.post(`${this.prefix}/query`, params);
  return res;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  attachMethod,
  cancelMethod,
  checkMethod,
  checkoutMethod,
  openBillingPortalMethod,
  queryMethod,
  setupPaymentMethod,
  trackMethod
});
